odl/test/trafos/fourier_test.py

```diff
@@ -858,5 +859,74 @@ def fhat(x):
     assert np.allclose(ft_f_n, fhat(recip_n.coord_vectors[0]))
 
 
+# ---- Non-uniform Fourier Transform ---- #
+def test_non_uniform_fourier_trafo_2d_call():
+    im_size = 512
+    shape = [im_size] * 2
+    samples = np.array(np.where(np.random.normal(size=shape) >= 0), dtype=float).T
+    samples /= 2.0 *  im_size
+    image = np.random.normal(size=shape)
+    nfft = NonUniformFourierTransform(shape=shape, non_uniform_samples=samples)
+    nfft(image)
+
+def test_non_uniform_fourier_adj_trafo_2d_call():
+    im_size = 512
+    shape = [im_size] * 2
+    samples = np.array(np.where(np.random.normal(size=shape) >= 0), dtype=float).T
+    samples /= 2.0 * im_size
+    nfft_coeffs = np.random.normal(size=(len(samples),))
+    nfft_adj = NonUniformFourierTransformAdjoint(
+        shape=shape,
+        non_uniform_samples=samples,
+    )
+    nfft_adj(nfft_coeffs)
+
+def test_non_uniform_fourier_trafo_1d_call():
+    sig_size = 512
+    shape = [sig_size]
+    samples = np.array(np.where(np.random.normal(size=shape) >= 0), dtype=float).T
+    samples /= 2.0 *  sig_size
+    image = np.random.normal(size=shape)
+    nfft = NonUniformFourierTransform(shape=shape, non_uniform_samples=samples)
+    nfft(image)
+
+def test_non_uniform_fourier_adj_trafo_1d_call():
+    sig_size = 512
+    shape = [sig_size]
+    samples = np.array(np.where(np.random.normal(size=shape) >= 0), dtype=float).T
+    samples /= 2.0 * sig_size
+    nfft_coeffs = np.random.normal(size=(len(samples),))
+    nfft_adj = NonUniformFourierTransformAdjoint(
+        shape=shape,
+        non_uniform_samples=samples,
+    )
+    nfft_adj(nfft_coeffs)
+
+
+def test_non_uniform_fourier_trafo_1d_res():
+    sig_size = 512
+    shape = [sig_size]
+    samples = np.array(np.where(np.ones(shape) >= 0), dtype=float).T
+    samples /= sig_size
+    samples -= 0.5
+    sig = np.random.normal(size=shape)
+    nfft = NonUniformFourierTransform(shape=shape, non_uniform_samples=samples)
+    res_nfft = nfft(sig)
+    res_np_fft = np.fft.fftshift(np.fft.fft(np.fft.fftshift(sig), norm="ortho"))
+    assert np.allclose(res_nfft, res_np_fft)
+
+def test_non_uniform_fourier_trafo_2d_res():
+    im_size = 512
+    shape = [im_size] * 2
+    samples = np.array(np.where(np.ones(shape) >= 0), dtype=float).T
+    samples /= im_size
+    samples -= 0.5
+    image = np.random.normal(size=shape)
+    nfft = NonUniformFourierTransform(shape=shape, non_uniform_samples=samples)
+    res_nfft = nfft(image)
+    res_np_fft = np.fft.fftshift(np.fft.fft2(np.fft.fftshift(image), norm="ortho"))
+    assert np.allclose(res_nfft, res_np_fft.flatten())
```

Comment:
Remark btw on this. I stuck to the form used everywhere else in the unit tests, but I stumbled upon [this](https://github.com/numpy/numpy/issues/9988) and I am wondering whether we should start using the more correct form `np.testing.assert_allclose`.